src/client/store/ui/geo/slice.ts

```diff
@@ -174,9 +182,57 @@ export const geoSlice = createSlice({
       state.forestOptions.agreementLayerSelected = true
       state.forestOptions.agreementLevel = agreementLevel
     },
-    setCustomAssetId: (state, { payload }: PayloadAction<string>) => {
+    setCustomForestAssetId: (state, { payload }: PayloadAction<string>) => {
       state.forestOptions.customAssetId = payload
     },
+    toggleProtectedAreaLayer: (state, { payload }: PayloadAction<ProtectedAreaKey>) => {
+      const i = state.protectedAreasOptions.selected.findIndex((key) => key === payload)
+      if (i === -1) {
+        delete state.protectedAreasOptions.failedLayers[payload] // In case the loading failed and it is manually re-tried
+        state.protectedAreasOptions.selected.push(payload)
+      } else {
+        state.protectedAreasOptions.selected.splice(i, 1)
+        // Reset opacity
+        delete state.protectedAreasOptions.opacity[payload]
+      }
+    },
+    setProtectedAreasLayers: (
+      state,
+      { payload: { sources, opacity } }: PayloadAction<{ sources: ProtectedAreaKey[]; opacity?: number }>
+    ) => {
+      state.protectedAreasOptions.selected = sources
+      state.protectedAreasOptions.opacity = {}
+
+      if (opacity !== undefined) {
+        state.protectedAreasOptions.selected.forEach((key) => {
+          state.protectedAreasOptions.opacity[key] = opacity
+        })
+      }
+    },
+    setProtectedAreaLayerOpacity: (
+      state,
+      { payload: { key, opacity } }: PayloadAction<{ key: string; opacity: number }>
+    ) => {
+      state.protectedAreasOptions.opacity[key] = opacity
+    },
+    setProtectedAreaGlobalOpacity: (state, { payload }: PayloadAction<number>) => {
+      state.protectedAreasOptions.selected.forEach((layerKey) => {
+        state.protectedAreasOptions.opacity[layerKey] = payload
+      })
+    },
+    resetProtectedAreaLayersStates: (state) => {
+      state.protectedAreasOptions.fetchedLayers = initialState.protectedAreasOptions.fetchedLayers
+      state.protectedAreasOptions.pendingLayers = initialState.protectedAreasOptions.pendingLayers
+      state.protectedAreasOptions.failedLayers = initialState.protectedAreasOptions.failedLayers
+    },
+    resetSingleProtectedAreaLayerStates: (state, { payload }: PayloadAction<ProtectedAreaKey>) => {
+      delete state.protectedAreasOptions.fetchedLayers[payload]
+      delete state.protectedAreasOptions.pendingLayers[payload]
+      delete state.protectedAreasOptions.failedLayers[payload]
+    },
+    setCustomProtectedAreaAssetId: (state, { payload }: PayloadAction<string>) => {
+      state.protectedAreasOptions.customAssetId = payload
+    },
```

Comment:
This file is becoming too big , that means there's either common logic to refactor or we should split this into several files ... what do you think ?

Reply:
Yes, I agree it is getting too big. The logic for the layers is similar but not always the same, so generalizing that logic would end up in a lot of extra code handling edge cases. I think it should better be split. Is it okay to do it in the upcoming Burned Areas PR?

Reply:
let's think how to refactor this later on?

Reply:
Yes, splitting it into files is a good start, but there should be a better way to handle the layers. We can take a look into that later.

Reply:
well, the best way would be to avoid duplicated code.